cli/src/lib.rs

```diff
@@ -624,6 +634,118 @@ pub fn entry(opts: Opts) -> Result<()> {
     }
 }
 
+/// Override the toolchain from `Anchor.toml`.
+///
+/// Returns previous versions.
+fn override_toolchain(cfg_override: &ConfigOverride) -> Result<ToolchainConfig> {
+    let mut previous_versions = ToolchainConfig::default();
+
+    let cfg = Config::discover(cfg_override)?;
+    if let Some(cfg) = cfg {
+        fn get_installed_version(cmd_name: &str) -> Result<String> {
+            let output: std::process::Output = std::process::Command::new(cmd_name)
+                .arg("--version")
+                .output()?;
+            let output_version = std::str::from_utf8(&output.stdout)?;
+            let version = Regex::new(r"(\d+\.\d+\.\S+)")
+                .unwrap()
+                .captures_iter(output_version)
+                .next()
+                .unwrap()
+                .get(0)
+                .unwrap()
+                .as_str()
+                .to_string();
+
+            Ok(version)
+        }
+
+        if let Some(solana_version) = &cfg.toolchain.solana_version {
+            let current_version = get_installed_version("solana")?;
+            if solana_version != &current_version {
+                // We are overriding with `solana-install` command instead of using the binaries
+                // from `~/.local/share/solana/install/releases` because we use multiple Solana
+                // binaries in various commands.
+                let exit_status = std::process::Command::new("solana-install")
```

Comment:
Ideally we can only change the symlink `~/.local/share/solana/install/active_release` when already installed, this would reinstall it every time, possibly annoying but i don't know if switching that symlink. Or maybe `solana` has this switcher somewhere.